Add `Button` functional tests.

lzhoucs opened the pull request:
**Type:** functional tests

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

<!--
Our bots should ensure:

* [ ] All contributors have signed a CLA
* [ ] The PR passes CI testing
* [ ] Code coverage is maintained
* [ ] The PR has been reviewed and approved
-->

Resolves #149

dylans commented on src/button/tests/functional/Button.ts:
```diff
@@ -0,0 +1,73 @@
+	'button should be visible'(this: any) {
+		return getPage(this.remote)
+			.findByCssSelector('button:first-of-type')
+		// as per mwistrand, `isDisplayed` is broken in Safari 10, commented out for now
```
Could we add a link to a Selenium driver issue listed somewhere so we can come back and fix this later?

dylans replied:
Also comment indentation is off

lzhoucs replied:
I just added the Selenium issue link and fixed the indentation. Please let me know if there's any other issues. Thanks!

lzhoucs commented:
Yeah. It is the known Safari issue on browserstack for today: `NoSuchDriver`.

tomdye commented on src/button/tests/functional/Button.ts:
```diff
@@ -0,0 +1,73 @@
+
+	'button should be visible'(this: any) {
+		return getPage(this.remote)
+			.findByCssSelector('button:first-of-type')
```
Please change all `findByCssSelector` calls to use the generate `css-module` class names by importing the appropriate `m.css` file.

lzhoucs replied:
Thanks for the review Tom. I looked at https://github.com/dojo/widgets/pull/166/commits/6612379717688a6b3300fccd97d2546c32233846 where we changed to use `css-module` class to access the internals of a widget, and this approach makes sense to me. However, in my case here where the widget structure is simple and I am not trying to exercise the internal components of the widget, but the widget itself - I am trying to locate the first widget on the page (there's several button widgets in the example), so I am not sure if `css-module` class would be any help locating what I am looking for. For example, by using `css-module`, it seems I still have to do something like this: ``.findByCssSelector(`button.${css.root}`:first-of-type)``. That said, the example page probably should have included ids for each widget for external tests like this one.

tomdye replied:
I see what you are saying @lzhoucs, however you are assuming that the tag will be `button`. This is not an assumption that you can make in all cases. More importantly, these tests are also to help with dojo 2 adopters and it's important that we consistently adopt `the right way` to deliver our code and tests. As per our `css-modules` approach, no css selector at any point should ever reference the tagname.

lzhoucs replied:
I agree that tag name shouldn't be used. I've added a wrapper div with id for each widget in the example, together with `css-module` class to locate the widget. Please take another look when you get chance, @tomdye, thanks!

tomdye replied:
Much better, although I'd have used a class over an id. 😉

lzhoucs replied:
I think id is not always bad. In the example page, each example is unique. There will always be one `example-1` on the page. And I am not trying to consistently style multiple examples, but access each of them individually.

tomdye commented on src/button/tests/functional/Button.ts:
```diff
@@ -0,0 +1,73 @@
+	'button should be visible'(this: any) {
+		return getPage(this.remote)
+			.findByCssSelector('button:first-of-type')
+		// `isDisplayed` is broken in Safari 10, commented out for now. See https://github.com/SeleniumHQ/selenium/issues/3029
```
Rather than leaving this in, can we please remove it in a separate commit? Then it can be added back in again as and when it's appropriate to do so. I think it's better not to have commented out code in our master branch.

lzhoucs replied:
Dylan suggested a slightly different approach: https://github.com/dojo/widgets/pull/168#discussion_r116070666, so that's why the comment is there. I've removed the commented code, however I do have a concern that it might get lost over time once the issue with Safari is fixed.

lzhoucs commented:
@tomdye , thanks for the review. I've addressed your feedback. Please take another look when you get chance and let me know if there's any other issues. Thanks!
